lib/OrdinaryDiffEqRosenbrock/src/rosenbrock_perform_step.jl

```diff
@@ -629,53 +618,35 @@ end
         return nothing
     end
 
-    linsolve_tmp = integrator.fsalfirst + dtd1 * dT
-
-    k1 = _reshape(W \ -_vec(linsolve_tmp), axes(uprev))
-    integrator.stats.nsolve += 1
-    u = uprev # +a21*k1 a21 == 0
-    # du = f(u, p, t+c2*dt) c2 == 0 and a21 == 0 => du = f(uprev, p, t) == fsalfirst
-
-    if mass_matrix === I
-        linsolve_tmp = integrator.fsalfirst + dtd2 * dT + dtC21 * k1
-    else
-        linsolve_tmp = integrator.fsalfirst + dtd2 * dT + mass_matrix * (dtC21 * k1)
-    end
-
-    k2 = _reshape(W \ -_vec(linsolve_tmp), axes(uprev))
+    k = Vector{typeof(uprev)}(undef, length(A))
+    linsolve_tmp = integrator.fsalfirst + dtd[1] * dT
+    k[1] = _reshape(W \ -_vec(linsolve_tmp), axes(uprev))
     integrator.stats.nsolve += 1
-    u = uprev + a31 * k1 + a32 * k2
-    du = f(u, p, t + c3 * dt)
-    integrator.stats.nf += 1
 
-    if mass_matrix === I
-        linsolve_tmp = du + dtd3 * dT + dtC31 * k1 + dtC32 * k2
-    else
-        linsolve_tmp = du + dtd3 * dT + mass_matrix * (dtC31 * k1 + dtC32 * k2)
-    end
+    for i in 2:eachindex(A)
```

Comment:
I don't think this is right since `eachindex` of a Matrix will loop over the scalars. I think you want `for i in 2:size(A, 2)`

Reply:
you also probably want an `@assert` that `size(A,1)==size(A,2)==length(k)` so the compiler can know that all the sizes match up.